backend/src/main/java/com/github/jonashonecker/backend/ticket/EmbeddingService.java

```diff
@@ -0,0 +1,52 @@
+package com.github.jonashonecker.backend.ticket;
+
+import com.github.jonashonecker.backend.ticket.domain.embedding.EmbeddingRequestDTO;
+import com.github.jonashonecker.backend.ticket.domain.embedding.EmbeddingResponseDTO;
+import com.github.jonashonecker.backend.ticket.domain.ticket.TicketWithTitleAndDescription;
+import com.github.jonashonecker.backend.ticket.exception.EmbeddingFailedException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClient;
+
+import java.util.List;
+
+@Service
+public class EmbeddingService {
+
+    private final RestClient restClient;
+
+    public EmbeddingService(
+            @Value("${app.openai-api-key}") String apiKey,
+            @Value("${app.openai-embedding-baseUrl}") String baseUrl
+    ) {
+        this.restClient = RestClient.builder()
+                .baseUrl(baseUrl)
+                .defaultHeader("Authorization", "Bearer " + apiKey)
+                .build();
+    }
+
+    public List<Double> getEmbeddingVectorForTicket(TicketWithTitleAndDescription ticket) {
+        return getEmbeddingVector("<h1>" + ticket.title() + "</h1>" + ticket.description());
```

Comment:
Do you really want to return html syntax?

Reply:
It's not returning html, it's returning a an array of doubles (an embedding vector). I'm basically just combining the title and description into one string to have them both in the embedding since most users write almost all of their descriptions about an issue into the ticket title.